src/main/gui/main-window.ts

```diff
@@ -360,10 +360,15 @@ export const createMainWindow = async (args: OpenArguments) => {
         const backendStatusProgressSlice = SubProgress.slice(progressController, 0.5, 0.95);
         sse.addEventListener('backend-status', (e: MessageEvent<string>) => {
             if (e.data) {
-                const data = JSON.parse(e.data) as { message: string; percent: number };
+                const data = JSON.parse(e.data) as {
+                    message: string;
+                    progress: number;
+                    statusProgress?: number;
+                };
```

Comment:
I like my code [super dry](https://www.youtube.com/watch?v=fZC0Utqi-Ro).
```suggestion
                const data = JSON.parse(e.data) as BackendEventMap['backend-status'];
```